src/modules/patients/patients.controller.ts

```diff
@@ -40,20 +43,62 @@ export class PatientsController {
     @Body() dto: CreatePatientDto,
   ): Promise<IServerResponse<Patient>> {
     const patient = await this.patientsService.addPatient(dto);
+
     return { statusCode: HttpStatus.OK, data: patient };
   }
 
   @Get()
-  @ApiOperation({ summary: 'Get latest patients' })
+  @ApiOperation({ summary: 'Get latest patients or search by name' })
   @ApiResponse({
     status: 201,
-    description: 'Get latest patients or search by name',
+    description: 'List of the patients found',
   })
   async getAll(
     @Query() searchOptions: PatientSearchOptionsDto,
   ): Promise<PatientsRes> {
     const response = await this.patientsService.getPatients(searchOptions);
     if (!response) throw new NotFoundException('There are no patients!');
+
     return response;
   }
+
+  @Get('/:id')
+  @ApiOperation({ summary: "Get full patient's info" })
+  @ApiResponse({
+    status: 201,
+    description: 'Patient information with appointment notes',
+  })
+  @ApiResponse({
+    status: 401,
+    description: 'Patient not found',
+  })
+  async getPatientWithNotes(
+    @Param() params: { id: number },
+  ): Promise<IServerResponse<Patient>> {
+    const patient = await this.patientsService.getPatientById(params.id);
+
+    return {
+      statusCode: HttpStatus.OK,
+      data: patient,
+    };
+  }
+
+  @Patch('update/:id')
+  @ApiOperation({ summary: "Update patient's info" })
+  @ApiResponse({
+    status: 201,
+    description: 'Patient information updated successfully',
+  })
+  @ApiResponse({
+    status: 401,
+    description: 'Patient not found',
+  })
+  async confirm(
+    @Param() params: { id: number },
+    @Body() dto: UpdatePatientDto,
+  ): Promise<IServerResponse<UpdatePatientDto>> {
+    const patient = await this.patientsService.updatePatient(+params.id, dto);
```

Comment:
instead of doing +params.id all the time you can create a general rule for it

Reply:
It was working without "+" too, forgot to remove it.
Not sure what do you meant by general rule, but added class-transformer @Type(() => Number) to validation